apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -201,14 +202,35 @@ session_add_iterator(DSSessionId, TopicFilter) ->
         end),
     Res.
 
+-spec session_get_iterator_id(session_id(), emqx_topic:words()) ->
+    {ok, iterator_id()} | {error, not_found}.
+session_get_iterator_id(DSSessionId, TopicFilter) ->
+    IteratorRefId = {DSSessionId, TopicFilter},
+    case mnesia:dirty_read(?ITERATOR_REF_TAB, IteratorRefId) of
+        [] ->
+            {error, not_found};
+        [#iterator_ref{it_id = IteratorId}] ->
+            {ok, IteratorId}
+    end.
+
 %% @doc Called when a client unsubscribes from a topic. Returns `true'
 %% if the session contained the subscription or `false' if it wasn't
 %% subscribed.
 -spec session_del_iterator(session_id(), emqx_topic:words()) ->
     {ok, boolean()} | {error, session_not_found}.
-session_del_iterator(_SessionId, _TopicFilter) ->
-    %% TODO
-    {ok, false}.
+session_del_iterator(DSSessionId, TopicFilter) ->
+    IteratorRefId = {DSSessionId, TopicFilter},
+    {atomic, Res} =
+        mria:transaction(?DS_SHARD, fun() ->
+            case mnesia:read(?ITERATOR_REF_TAB, IteratorRefId, write) of
+                [] ->
+                    {ok, _WasSubscribed = false};
+                [#iterator_ref{}] ->
+                    ok = mnesia:delete(?ITERATOR_REF_TAB, IteratorRefId, write),
+                    {ok, _WasSubscribed = true}
+            end
```

Comment:
I think @keynslug need to weigh on this, since write amplification is a real issue to be considered. Ideally we want to support clients that subscribe to large number of topics (in thousands), so it's a real issue.
For now, I think we can overlook this part and move on.

Reply:
> TBH I don't see why it's important to respond with WasSubscribed flag,

I was trying to adhere the [previously defined typespec and doc](https://github.com/emqx/emqx/blob/8b3a9072f970fed8e2a79fba65ac5093d7637013/apps/emqx_durable_storage/src/emqx_ds.erl#L204-L208) for this function.  But, at this moment, it doesn't seem to be useful, indeed.

I'll then add a field to our `#session{}` record to keep track of the iterators, then.  That way, we can avoid the lookup before performing the multicall.